### ISSUE TITLE
Sharing Custom Swagger UI Content Between ASP.NET Core Apps

### ISSUE THREAD

**Issue by RehanSaeed** [domaindrivendev/Ahoy#150]
I have customized a customized Swashbuckle UI using some custom HTML, CSS and image files in the following folders of my ASP.NET Core site:

```
wwwroot
  swagger
    ui
      images
        background.jpg
        favicon.ico
        logo_small.png
    index.html
    skin.css
```

How can I share these files in a separate class library so that multiple sites can use the same UI.

**Comment by domaindrivendev** [comment]
Essentially, you're asking how to embed static assets in a class library and then serve them up through an ASP.NET Core application which references that library. It seems like more of an ASP.NET Core question and not something specific to SB.

With that said, I've had to do exactly that with Swashbuckle, serving the swagger-ui assets out of the Swashbuckle.SwaggerUi class library. So, I'm happy to share the approach I took:

Firstly, I specify the files to be embedded into your class library at runtime as can be seen in Swashbuckle.SwaggerUi:

https://github.com/domaindrivendev/Ahoy/blob/master/src/Swashbuckle.SwaggerUi/project.json#L7

Then I leverage the built-in, file server middleware to serve these files under a given request path.

https://github.com/domaindrivendev/Ahoy/blob/master/src/Swashbuckle.SwaggerUi/Application/SwaggerUiBuilderExtensions.cs#L33

In your case, you would want to insert this just before the call to __UseSwaggerUi__. This way, I believe it will try serve the requests from your embedded folder first and if it can't, delegate to the Swashbuckle SwaggerUI middleware.
